File: src/designpatterns/builder/director.cpp
#include "concrete_builder.hpp"

class Director {
public:
    static void constructMeal(Builder& builder) {
        builder.addStarter().addMain().addDessert().addDrink();
    }
};

int main() {
    VeganBuilder veganBuilder;

    Director::constructMeal(veganBuilder);
    std::unique_ptr<Meal> veganMeal = veganBuilder.build();
    
    return 0;
}

File: src/designpatterns/builder/BuilderJava/Director.java
package designpatterns.builder.BuilderJava;

public class Director {
    
    static void constructMeal(Builder builder) {
        builder.addStarter().addMain().addDessert().addDrink();
    }

    public static void main(String[] args) {
        Builder veganBuilder = new VeganBuilder();
        Director.constructMeal(veganBuilder);

        Meal veganMeal = veganBuilder.build();
        System.out.println(veganMeal);
    }
}